### ISSUE TITLE
Global namespace pollution by installing `tests`

### ISSUE THREAD

**Issue by quantumkoen** [Bachmann1234/marshmallow-polyfield#23]
I've been banging my head around some issues I had with running test-cases of one of our projects installed with `pip install -e .`. After a lot of debugging, the problem is that marshmallow-polyfield installs a package 'tests' in the global namespace which conflicts with our use of a directory `src/tests`.

Long story short, with `pip install -e .`, our `src/` dir is appended at the end of pythons `sys.path`. However, since marshmallow-polyfield's `tests` package is installed in the `site-packages` dir, which comes before our `src/` dir in `sys.path`, we would be getting errors about not being able to import `tests.test_foo`.

Is it really necessary for marshmallow-polyfield to install something so generic as `tests` in the global namespace?

**Comment by Bachmann1234** [comment]
probably not. I suspect we could just remove it from the setup.py without consequence. why dont you give it a shot and report back?

**Comment by quantumkoen** [comment]
@Bachmann1234 why not? probably because i've spent hours pulling my hair out to find out why our unit tests were broken and am now calling it a day. As a sidenote, I don't even know what marshmallow-polyfield is and what someone would use it for, it's just a dependency of a dependency of a dependency for us. I would not know how to validate whether removing it from `setup.py` breaks things or not. But i'll try tomorrow, and report back whether our dependencies fall over or not.

**Comment by Bachmann1234** [comment]
I mean "It is probably not necessary to publish tests in setup.py" 

I was suggesting trying to remove it and reporting how it goes for you

**Comment by Bachmann1234** [comment]
Sorry, im a bit distracted right now. Ill try something this week when I get a moment.

**Comment by Bachmann1234** [comment]
try https://pypi.org/project/marshmallow-polyfield/5.5/ and see if that behaves better for you

**Comment by quantumkoen** [comment]
@Bachmann1234 thanks for your help, with 5.5 our issue with the test case imports is solved, however I now need to get the next one up the chain (the dependency of the dependency) to update, they've got the version of marshmallow-polyfield pinned to 3.2 it seems. I tried running with 5.5, but there are apparently incompatibilities because it gave me a whole lot of exceptions with very deep stacktraces that currently don't mean much to me. I need to investigate further.

But at least, as far as marshmallow-polyfield is concerned, I believe the issue has been addressed satisfactory! Thanks again for the swift response!
